src/Oscoin/Protocol/Sync.hs

```diff
@@ -0,0 +1,393 @@
+{-# LANGUAGE DataKinds            #-}
+{-# LANGUAGE UndecidableInstances #-}
+module Oscoin.Protocol.Sync
+    (
+    -- * Types
+      Sync(..)
+    , SyncError(..)
+    , Timeout(..)
+    , ActivePeer
+    , ActivePeers
+    , SyncEvent(..)
+    , SyncContext(..)
+    , DataFetcher(..)
+    , ProtocolRequest(..)
+    , SProtocolRequest(..)
+    , ProtocolResponse
+    , Range(..)
```

Comment:
Does this have to be exported? If so it would be good to prefix the accessors, like `rangeStart`, `rangeEnd`

Reply:
Perhaps also naming it `HeightRange` :man_shrugging:

Reply:
It is useful to have it exported for testing, but sure enough we can prefix the accessors.